QA/PoroDevTest/src/main/java/common/ui_setup/pom_setup/PoroDevPom/LoginPage.java

```diff
@@ -1,55 +1,74 @@
 package common.ui_setup.pom_setup.PoroDevPom;
 
-import common.ui_setup.SetupConstants;
 import common.ui_setup.pom_setup.BasePage;
 import org.openqa.selenium.WebDriver;
 import org.openqa.selenium.WebElement;
 import org.openqa.selenium.support.FindBy;
-import java.io.IOException;
+import org.testng.Assert;
 
 public class LoginPage extends BasePage {
 
     public LoginPage(WebDriver driver) {
         super(driver);
-        driver.get(SetupConstants.BASE_URL);
+    }
+
+    public LoginPage(WebDriver driver, String url) {
+        super(driver);
+        driver.get(url);
     }
 
     //element section
-    @FindBy(id = "basic_email")
-    WebElement emailField;
-    @FindBy(id = "basic_password")
-    WebElement passField;
+    @FindBy(name = "email")
+    WebElement we_emailInputField;
+    @FindBy(name = "password")
+    WebElement we_passInputField;
     @FindBy(className = "ant-btn-primary")
-    WebElement loginButton;
+    WebElement we_loginButton;
     //ASSERTION ELEMENTS
-    @FindBy(xpath = "//div[contains(text(),'Please input your username')]")
-    public WebElement emailErrorMsg;
-    @FindBy(xpath = "//div[contains(text(),'Please input your password')]")
-    public WebElement passErrorMsg;
-    @FindBy(xpath = "//span[contains(text(),'Request failed with status code 401')]")
-    public WebElement errorAfterBadCredentials;
-    @FindBy(className = "hvVmBA")
-    public WebElement userFirstNameAfterLogin;
+    @FindBy(xpath = "//span[contains(text(),'Email is required')]")
+    WebElement we_email_requiredMsg;
+    @FindBy(xpath = "//span[contains(text(),'Password is required')]")
+    WebElement we_pass_requiredMsg;
+    @FindBy(xpath = "//span[contains(text(),'Invalid email')]")
+    WebElement we_email_invalidMsg;
+    @FindBy(xpath = "//span[contains(text(),'Wrong password')]")
+    WebElement we_pass_invalidMsg;
     @FindBy(xpath = "//span[contains(text(),'Successful login')]")
-    public WebElement successfulLogin;
+    WebElement we_successfulLogin;
+    @FindBy(className = "ant-message")
+    WebElement we_badRequestError;
 
-    //methods section
-    public void enterEmail(String email) {
-        elementControl.sendText(emailField, email);
+    //functional methods section
+
+    public void login(String email, String password) throws InterruptedException {
+        BasePage.sendText(we_emailInputField,email);
+        BasePage.sendText(we_passInputField, password);
+        BasePage.clickElement(we_loginButton);
+        Thread.sleep(2000);
```

Comment:
Bad practice. You shouldn't use Thread.sleep() anywhere in your project. Use Explicit wait instead.

Reply:
Got it